utils.go

```diff
@@ -111,3 +111,29 @@ func (l *limitedReadCloser) Read(buf []byte) (int, error) {
 func (l *limitedReadCloser) Close() error {
 	return l.rc.Close()
 }
+
+func GetContentLength(reader io.Reader) (int64, error) {
+	contentLength := int64(0)
```

Comment:
zero is a valid length for some transfers, this should probably return `-1` to indicate the length is unknown or return additional `bool`